master/internal/api_experiment.go

```diff
@@ -1916,3 +1923,158 @@ func (a *apiServer) GetModelDefFile(
 	}
 	return &apiv1.GetModelDefFileResponse{File: file}, nil
 }
+
+func (a *apiServer) SearchExperiments(ctx context.Context, req *apiv1.SearchExperimentsRequest) (*apiv1.SearchExperimentsResponse, error) {
+	resp := &apiv1.SearchExperimentsResponse{}
+	var experiments []*experimentv1.Experiment
+	var trials []*trialv1.Trial
+	experimentQuery := db.Bun().NewSelect().
+		Model(&experiments).
+		ModelTableExpr("experiments as e").
+		Apply(getExperimentColumns)
+
+	var err error
+	resp.Pagination, err = runPagedBunExperimentsQuery(ctx, experimentQuery, int(req.Offset), int(req.Limit))
+	if err != nil {
+		return nil, err
+	}
+
+	if err = a.enrichExperimentState(experiments...); err != nil {
+		return nil, err
+	}
+
+	// get the best trial associated with the experiment.
+
+	// don't query for experiments twice
+	experimentValues := db.Bun().NewValues(&experiments)
+
+	// extract config info from experiments and associate with trials
+	searcherInfoQuery := db.Bun().NewSelect().
+		Table("ex").
+		ColumnExpr("ex.config->'searcher'->>'metric' AS metric_name").
+		ColumnExpr("(SELECT CASE WHEN coalesce((ex.config->'searcher'->>'smaller_is_better')::boolean, true) THEN 1 ELSE -1 END) as sign").
+		ColumnExpr("trials.id AS trial_id").
+		Join("JOIN trials ON trials.experiment_id = ex.id")
+
+	// get info for best/latest validation for best trial
+	validationsQuery := db.Bun().NewSelect().
+		Table("validations").
+		Column("id").
+		ColumnExpr("validations.trial_id as trial_id").
+		Column("total_batches").
+		ColumnExpr("proto_time(end_time) AS end_time").
+		ColumnExpr("json_build_object('avg_metrics', metrics->'validation_metrics') AS metrics").
+		ColumnExpr("metrics->'num_inputs' AS num_inputs").
+		ColumnExpr("((metrics->'validation_metrics'->>(si.metric_name))::float8 * si.sign) AS signed_searcher_metric").
+		ColumnExpr("row_number() OVER(PARTITION BY validations.trial_id ORDER BY end_time DESC NULLS LAST) AS latest_rank").
```

Comment:
i think we want to filter for validation `state = 'COMPLETED'` here 

also, i think better to order by `total_batches` since the index is `(trial_id, total_batches, ...)` and we are already filtering on `trial_id`

Reply:
the query here doesn't do either, but i do think your suggestions are good: https://github.com/determined-ai/determined/blob/ad5f6e5f91396d527e8dd67d86da38366170bb74/master/static/srv/proto_get_trials_plus.sql#L59-L75

quick q: is it possible for a trial to have multiple validations at the same batch size?

Reply:
in the db, it looks like the unique constraint is `(trial_id, total_batches, trial_run_id)` so I guess there is that same question we had for steps. I think it would probably fine to throw `trial_run_id` in the `ORDER BY DESC` since presumably it's increasing. I think it's rare enough that there would be more than one anyways.

Reply:
nice!